### PR TITLE
feat(shorebird_code_push): track support

### PR DESCRIPTION
**Pull request by felangel**
## Description

Add track support

https://github.com/user-attachments/assets/e4ed1e87-125f-4f4f-99ef-1e0fbcf3075b

## Type of Change

- [X] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: library/src/updater.rs [hunk 1]
```diff
@@ -243,9 +243,9 @@ pub fn should_auto_update() -> anyhow::Result<bool> {
 }
 
 /// Synchronously checks for an update and returns true if an update is available.
-pub fn check_for_update() -> anyhow::Result<bool> {
+pub fn check_for_update(channel: *const char) -> anyhow::Result<bool> {
     let (request, url, request_fn) = with_config(|config| {
-        // Get the required info to make the request.
+        // TODO: use the channel in the patch check request
```
Comment (felangel) [inline library/src/updater.rs:248]: Need to use the channel here when making the request

File: shorebird_code_push/example/lib/main.dart [hunk 2]
```diff
@@ -31,6 +31,7 @@ class _MyHomePageState extends State<MyHomePage> {
   late final bool _isUpdaterAvailable;
   var _isCheckingForUpdates = false;
   Patch? _currentPatch;
+  UpdateTrack? _currentTrack;
```
Comment (felangel) [inline shorebird_code_push/example/lib/main.dart:34]: Need to also add a bool to track the initialization state of the track

File: shorebird_code_push/lib/src/updater.dart [hunk 3]
```diff
@@ -20,19 +21,24 @@ class Updater {
...
   int nextPatchNumber() => bindings.shorebird_next_boot_patch_number();
 
   /// Downloads the latest patch, if available.
   void downloadUpdate() => bindings.shorebird_update();
 
+  // New Methods added to support v2.0.0 of the Dart APIs //
+
+  /// Whether a new patch is available.
+  bool checkForUpdate({UpdateTrack? track}) =>
+      // TODO: convert the track to a channel and pass it to the bindings.
```
Comment (felangel) [inline shorebird_code_push/lib/src/updater.dart:35]: Need to transform the track to a channel and pass it to the bindings here.

File: shorebird_code_push/example/lib/main.dart (hunk 2 above)
Comment (bryanoltman) [reply shorebird_code_push/example/lib/main.dart:34]: "track the initialization state" = `hasUpdateTrackBeenSet`?

**Comment by bryanoltman** [comment]
I think this is ready for review. @eseidel and I spoke about adding a version reporting function to the C API, but that probably belongs in a separate PR

**Review by bryanoltman — APPROVED** [review]

**Comment by bryanoltman** [comment]
Verbal LGTM from @felangel on Discord

**Pull request merged**